src/v/model/transform.h

```diff
@@ -102,14 +102,15 @@ struct transform_offset_options
     //
     // When a timestamp is used, a timequery is used to resolve the offset for
     // each partition.
-    serde::variant<latest_offset, model::timestamp> position;
+    serde::variant<latest_offset, model::timestamp, model::offset> position;
```

Comment:
nit: offset_delta is a better fit? And please update the docs

Reply:
Ya I'm not exactly sure. I balked initially because the on-label use for offset_delta is translating between logical and physical offsets. It's all ints at the end of the day, so it felt a but better to use the more generic option. Will consider

Reply:
I think it's conceptually a `kafka::offset_delta` (rather than a `model::offset_delta` or `kafka|model::offset`) since we're adding it onto a kafka start/end offset.

Reply:
Near as I can tell `model::offset_delta` is explicitly meant to hit overloads that perform the model::offset <-> kafka::offset conversion automatically, e.g.

```c++
/// \brief conversion from kafka offset to redpanda offset
inline constexpr model::offset
operator+(kafka::offset o, model::offset_delta d) {
    return model::offset{o() + d()};
}
```
So yeah, I agree with Gellert. Maybe just add `kafka::offset_delta` for specifically the purpose of staying w/in `kafka` land.